packages/connect-node/src/node-universal-handler.spec.ts

```diff
@@ -63,50 +65,91 @@ describe("universalRequestFromNodeRequest()", function () {
     }
     it("should abort request signal with ConnectError and Code.Canceled for NO_ERROR", async function () {
       await request(http2.constants.NGHTTP2_NO_ERROR);
-      expect(universalRequestSignal).toBeInstanceOf(AbortSignal);
-      expect(universalRequestSignal?.aborted).toBeTrue();
-      expect(universalRequestSignal?.reason).toBeInstanceOf(Error);
-      if (universalRequestSignal?.reason instanceof Error) {
-        expect(universalRequestSignal.reason.name).toBe("AbortError");
-        expect(universalRequestSignal.reason.message).toBe(
+      expect(serverRequest?.signal).toBeInstanceOf(AbortSignal);
+      expect(serverRequest?.signal.aborted).toBeTrue();
+      expect(serverRequest?.signal.reason).toBeInstanceOf(Error);
+      if (serverRequest?.signal.reason instanceof Error) {
+        expect(serverRequest.signal.reason.name).toBe("AbortError");
+        expect(serverRequest.signal.reason.message).toBe(
           "This operation was aborted"
         );
       }
```

Comment:
since `signal.reason` is typed `any`, the correct way to pass this information to the type system is with an `asserts` function:

```ts
function assertIsError(error: unknown): asserts error is Error {
  if (!(error instanceof Error)) {
    throw new Error(`input ${error} is not an instance of Error`)
  }
};
```

Then we can avoid the (usually noop) conditional in a test

```suggestion
  assertIsError(serverRequest?.signal.reason);
  expect(serverRequest.signal.reason.name).toBe("AbortError");
  expect(serverRequest.signal.reason.message).toBe(
    "This operation was aborted"
  );
```

Reply:
I would prefer this to be built into the test framework. Maintaining assertion functions as helpers is not ideal.